src/ion_data/mod.rs

```diff
@@ -0,0 +1,110 @@
+mod ion_eq;
+mod ion_ord;
+
+use crate::{Element, IonResult};
+use std::cmp::Ordering;
+use std::fmt::{Display, Formatter};
+
+use crate::element::{Annotations, Value};
+use delegate::delegate;
+pub use ion_eq::IonEq;
+pub(crate) use ion_ord::IonOrd;
+
+/// A wrapper for lifting "data" to "Ion data". Anything that implements [IonEq] can be converted
```

Comment:
The first sentence doesn't really indicate the point of the class IMNSHO.  Maybe something like:

> A wrapper for lifting Ion compatible data into using Ion oriented comparisons (versus the Rust value semantics).  This enables the default semantics to be what a Rust user expects for native values, but allows a user to opt-in to Ion's structural equivalence/order.

Reply:
I was hoping that you would have a better way of describing the purpose of this type. Thanks! 😉